test/puppetlabs/trapperkeeper/services/watcher/filesystem_watch_service_test.clj

```diff
@@ -385,3 +385,22 @@
        (is (logged?
             #"shutdown-on-error triggered because of exception"
             :error))))))
+
+(deftest ^:integration overflow-test
+  (let  [watch-dir  (fs/temp-dir "overflow")
+        results  (atom  [])
+        callback  (fn  [events]  (swap! results concat events))]
+    (with-app-with-config
+     app watch-service-and-deps  {}
+     (let  [service  (tk-app/get-service app :FilesystemWatchService)]
+       (watch! service watch-dir callback))
+     ;; From experimenting the buffer seems to fill at 512 events.
+     ;; However the number of events varies by operating system.
```

Comment:
If this passes reliably, that's cool.  If it starts to fail intermittently, or we wanted to make it more robust, we could change to create files indefinitely until an overflow event occurs.  But, I'm good with this approach for now.